### PR TITLE
Allow for respawn and erase verb to be used on offline players. Also minor  rewrite on respawn command and new erase command

### LINKED ISSUE #23796: Most admin verbs aren't available for offline/souless players/characters
## Description
Some admin verbs aren't available for offline players, despite still being useful. Useful verbs include: Erase, Open Admin Notes, etc. 

**Reproduction**
- Try to erase a player who logged off, you can't.
- Try to respawn a player who's offline via the verb.

**Screenshots**
offline
image
online
image

**Additional context**
Some of them are useless when the player is offline, such as "Subtle Message", and so make sense to not appear.

### PR DESCRIPTION
**Pull request by nikthechampiongr**
## About the PR

fixes #23796
This command makes the respawn and erase verbs available to be used on disconnected players.

Also localizes the respawn command and allows for it to accept userids.

## Why / Balance

admin qol

The erase command is added for the edgecase where an admin wants to erase someone, without banning them, and without being able to find an entity with their mind attached to it.

## Technical details

respawn command rewritten so that it can accept userids, was localized, and now uses [Dependency] where possible.

respawn verb is now available on disconnected players.

Erase now accepts a netuserid instead of a session.

As it turns out, the only reason the actor component was used for these is because  it was a convenient way to get the netuserid, however the player does not need to be logged in for these actions to work.

Adminhelp and subtle message verbs could also be made to not require an actor component I believe, however it would not make sense to do this on offline players. 

Spawn, and Clone verbs would require a db lookup which is async and verbs are not async. While I think I know a way to get around this it would be convoluted. Respawn is the most important verb for this anyways. 

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:

ADMIN:

- fix: Respawn verb is now available on players even after they have disconnected.
- fix: Erase command is now available on players even after they have disconnected.
- add: You can now erase people with the erase command
- tweak: Respawn command now also accepts user ids

### CODE REVIEW

**Comment by nikthechampiongr** [comment]
I am expanding the scope of this pr to most verbs as most of them appear to be actually possible without an actor component. Long live CE.

**Comment by nikthechampiongr** [comment]
Currently proceeding with just erase and respawn not needing an actor component as they are the only ones which are possible and make sense.

**Comment by nikthechampiongr** [comment]
nevermind one moment

File: Content.Server/Administration/Commands/EraseCommand.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            shell.WriteLine(Loc.GetString("cmd-erase-invalid-args"));
+            shell.WriteLine(Help);
+            return;
+        }
+
+        var located = await _locator.LookupIdByNameOrIdAsync(args[0]);
+
+        if (located == null)
+        {
+            shell.WriteLine(Loc.GetString("cmd-erase-player-not-found"));
```
Comment (metalgearsloth) [inline Content.Server/Administration/Commands/EraseCommand.cs:31]:
```suggestion
        {
            shell.WriteError(Loc.GetString("cmd-erase-player-not-found"));
```

File: Content.Server/Administration/Commands/EraseCommand.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            return;
+        }
+
+        if (_system.TryGetEntitySystem<AdminSystem>(out var adminSystem))
+        {
+            adminSystem.Erase(located.UserId);
+        }
```
Comment (metalgearsloth) [inline Content.Server/Administration/Commands/EraseCommand.cs:38]: Pretty sure pjb added a thing to inject these in some engine PR.

File: Content.Server/Administration/Commands/EraseCommand.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+  public override CompletionResult GetCompletion(IConsoleShell shell, string[] args)
+    {
+        if (args.Length != 1)
+            return CompletionResult.Empty;
+
+        var options = _players.Sessions.OrderBy(c => c.Name).Select(c => c.Name).ToArray();
+
+        return CompletionResult.FromHintOptions(options, "<PlayerIndex>");
```
Comment (metalgearsloth) [inline Content.Server/Administration/Commands/EraseCommand.cs:47]: Localise maybe?

File: Content.Server/GameTicking/Commands/RespawnCommand.cs [hunk 4]
```diff
@@ -1,62 +1,70 @@
...
+        [Dependency] private readonly IEntitySystemManager _systems = default!;
 
-        public void Execute(IConsoleShell shell, string argStr, string[] args)
+        public override string Command => "respawn";
+
+        public async override void Execute(IConsoleShell shell, string argStr, string[] args)
         {
+            if (!_systems.TryGetEntitySystem<GameTicker>(out var gameTicker) || !_systems.TryGetEntitySystem<MindSystem>(out var mind))
+                return;
+
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/Commands/RespawnCommand.cs:22]: As above.

File: Content.Server/GameTicking/Commands/RespawnCommand.cs [hunk 5]
```diff
@@ -1,62 +1,70 @@
...
             var player = shell.Player;
             if (args.Length > 1)
             {
-                shell.WriteLine("Must provide <= 1 argument.");
+                shell.WriteLine(Loc.GetString("cmd-respawn-invalid-args"));
                 return;
             }
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/Commands/RespawnCommand.cs:29]: WriteErrors in here too just can't highlight it all coz github things.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**